src/Microsoft.Graph.Cli.Core/Http/UriReplacementHandler.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph.Cli.Core.Http;
+
+/// <summary>
+/// Interface for making URI replacements.
+/// </summary>
+public interface IUriReplacement
+{
+    /// <summary>
+    /// Accepts a URI and returns a new URI with all replacements applied.
+    /// </summary>
+    /// <param name="original">The URI to apply replacements to</param>
+    /// <returns>A new URI with all replacements applied.</returns>
+    Uri? Replace(Uri? original);
+}
+
+/// <summary>
+/// Specialized replacement for /[version]/users/me with /[version]/me
+/// </summary>
+public struct MeUriReplacement : IUriReplacement
```

Comment:
I'd rather define a middleware handler in kiota http like we've done in the other languages (this was missed for dotnet during the GA)
This way it aligns with other languages, and the work will already be done for the next time we major rev the dotnet SDK.
https://github.com/microsoft/kiota-http-go/blob/main/url_replace_handler.go

Reply:
Ok. I'll move the handler, but there probably won't be a replacement implementation.

Reply:
Done. Will be dependent upon PR microsoft/kiota-cli-commons#87